### ISSUE TITLE
Using the Eye Highlighting feature with WP Recipe Maker inserts HTML markup in the content area.

### ISSUE THREAD

**Issue by rmarcano** [Yoast/YoastSEO.js#1420]
## What did you expect to happen?
When using the eye highlighter to highlight the "X% of the sentences contain a transition word or phrase, which is less than the recommended minimum of 30%." warning, I expected the content to be highlighted.

## What happened instead?
The eye highlighter for this warning seems to bring the HTML markup to the visual editor instead. See attached images for reference.

**Before highlighting the issues:**
Visual editor:
![before-visual](https://user-images.githubusercontent.com/1044883/36511497-44e1ecb8-173d-11e8-8e4f-0d1756d09b09.png)

Text editor:
![before](https://user-images.githubusercontent.com/1044883/36511513-51622a84-173d-11e8-86fc-3eab3a8f77bb.png)

**After highlighting the issues:**
Visual editor:
![after-visual](https://user-images.githubusercontent.com/1044883/36511528-5f79d5fe-173d-11e8-8912-c5e4f1be8ccc.png)

Text editor:
![after-text](https://user-images.githubusercontent.com/1044883/36511544-6cff7ce2-173d-11e8-87f9-b4a323b2ab5f.png)

## How can we reproduce this behavior?
1) Install WP Recipe Maker and Premium addons (might need to contact developer).
2) Create a recipe and insert it in to the post.
3) Create content until the transition word warning is triggered.
4) Try to highlight the issues in the content.

## Technical info

* Platform: WordPress 
* Platform version:4.9.4
* Yoast SEO version: 6.3.1
* Other plugins: WP Recipe Maker 2.1.1 and WP Recipe Maker premium bundle 2.0.2

**Comment by rmarcano** [comment]
Please inform the customer of conversation # 312139 when this conversation has been closed.

**Comment by smokinggunmedia** [comment]
Add Bootstrap Ventures of Recipe maker, sent email should be joining us soon

**Comment by BrechtVds** [comment]
I can supply you with a copy of WP Recipe Maker Premium for debugging. Just let me know how.

**Comment by smokinggunmedia** [comment]
Whats the plan guys?

**Comment by smokinggunmedia** [comment]
any movement on either side guys? @benvaassen @BrechtVds

**Comment by BrechtVds** [comment]
I'm waiting on instructions for where to deliver the Premium plugin, @smokinggunmedia 
I can't just attach the zip file in public here.

**Comment by smokinggunmedia** [comment]
@rmarcano where do you want the premium plug in sent?

**Comment by CarolineGeven** [comment]
@BrechtVds / @smokinggunmedia the premium plugin can be sent to caroline@yoast.com. Thank you.

**Comment by BrechtVds** [comment]
Sent.

**Comment by smokinggunmedia** [comment]
How we doing with this issue?

**Comment by smokinggunmedia** [comment]
@CarolineGeven hello.. did this get resolved, do you need access to my site https://www.justbeingvegan.com/

**Comment by smokinggunmedia** [comment]
can someone add the client to this thread
mailto:jalen@justbeingvegan.com

**Comment by smokinggunmedia** [comment]
Hey team, any movement on this? 

From the client
I tried to add more recipe posts last night but yoast is still not working with the Recipe maker. Has there been any further communication or resolution from them?

**Comment by benvaassen** [comment]
Apologies it took some time on our part to dive into this. 

I'm unable to reproduce the issue where it would bring the HTML markup to the visual editor instead, with WP Recipe Maker 2.3.0 and WP Recipe Maker premium bundle 2.0.2 and Yoast SEO 7.1. 

I am able to reproduce the issue with the eye highlight. None of them work when the shortcode for a recipe has been added, while they should work except for 1 of them (see: https://github.com/Yoast/YoastSEO.js/issues/1195). When I remove the recipe shortcode, all of them work (except for the 1). 

I'm going to assign @CarolineGeven to further investigate that issue.

**Comment by BrechtVds** [comment]
Premium 2.0.2 will not get loaded with WP Recipe Maker 2.3.0 because it requires an update. I've sent the latest Premium plugin file to caroline@yoast.com for testing.

**Comment by benvaassen** [comment]
Thanks for providing us with the latest plugin file. 

Testing with that version does trigger the added HTML markup to the visual editor, but only when I select the: `xx% of the sentences contain a transition word or phrase, which is less than the recommended minimum of 30%` highlight. The other highlight eyes simply don't work at all (have no effect). 

When I remove the recipe, the highlighting works as expected. Still going to leave this assigned to @CarolineGeven for further investigation.

**Comment by rmarcano** [comment]
Please inform the customer of conversation # 346697 when this conversation has been closed.

**Comment by moorscode** [comment]
This seems to be a problem that only occurs in combination with this specific plugin, thus indicating that the cause is most likely in their code.

We are more than willing to help find a solution together with the plugin author if the problem has been determined, we simply don't have the capacity to investigate this any further on our end.

If a concrete problem is detected that requires actions on our end, please open a new issue.

I'm really sorry for the long wait and this unsatisfying conclusion.

**Comment by smokinggunmedia** [comment]
@BrechtVds you working on fixing this yoast issue, i'm assuming you have client who use this tech.

**Comment by BrechtVds** [comment]
@moorscode can you point me to the part of your code responsible for the highlighting functionality?

**Comment by moorscode** [comment]
@BrechtVds 

The marks are being applied here: https://github.com/Yoast/wordpress-seo/blob/trunk/js/src/wp-seo-post-scraper.js#L117-L125

Which is using the functionality from the `tinyMCE`-decorator, which can be found here:
https://github.com/Yoast/wordpress-seo/blob/trunk/js/src/decorator/tinyMCE.js

**Comment by smokinggunmedia** [comment]
any movement on this?

**Comment by BrechtVds** [comment]
@smokinggunmedia it's on our todo list. I will update this issue if we find something. We're not familiar with this code though, so no guarantees.

**Comment by sumitg12** [comment]
Hello, I am running into the same issue with Yoast & WP Recipemaker Pro.   I *think* Yoast sees the recipe as one big paragraph and so gives two warnings / errors:
1.  "1 subheading has more than 300 words"
2. "1 of the paragraphs contains more than 150 words"

So, now every one of the recipes where we use WP Recipe Maker, is red with Needs improvement.

Also, the "eye highlight" feature does not work with WP Recipemaker.    

Then there is a big error.   If I click the "Eye highlight" for the warning around "Transition words", Yoast puts the HTML code into the "Visual" view of the Post.    Complete fail.

For example, here is one of the recipes: 
https://spicecravings.com/plum-jam
